### PR TITLE
Add python sms log handler

### PR DESCRIPTION
**Pull request by ikosenn**

### CODE REVIEW

File: .isort.cfg [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[settings]
+skip=.tox
+atomic=true
+multi_line_output=5
+known_standard_library=types
+known_third_party=pytest
+known_first_party=rest_framework_files
```
Comment (evansmurithi) [inline .isort.cfg]: Badilisha, hehe

File: sms_log_handler/providers/africastalking/base.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        Handles sending of the error message to the provider phone numbers
+
+        params:
+            phone_numbers: An array of phonenumbers to sms
+            message: This are the contents of the message that is being sent.
+
+        """
+        phone_numbers_csv = ','.join(phone_numbers)
+        gateway = AfricasTalkingGateway(self.key, self.secret)
+        gateway.send_message(phone_numbers_csv, message)
```
Comment (evansmurithi) [inline sms_log_handler/providers/africastalking/base.py]: There is no exception that will be raised if for example the phone numbers are wrong format?

File: sms_log_handler/providers/africastalking/gateway.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import urllib.request
+import urllib.parse
+import json
+
+
+class AfricasTalkingGatewayException(Exception):
+    pass
+
+
+class AfricasTalkingGateway:
```
Comment (evansmurithi) [inline sms_log_handler/providers/africastalking/gateway.py]: Why not import the `AfricasTalkingGateway` lib?

File: sms_log_handler/providers/base.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+class SMSProviderBase:
+    """
+    This is a base class that should be inherited by any new providers that
+    are brough abroad. It providers a common interface
```
Comment (evansmurithi) [inline sms_log_handler/providers/base.py]: `brought` typo

File: sms_log_handler/sms_handler.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        Contruct and format the mesage to be sent.
+
+        i.e
+        MODULE: sms_log_handler.sms_handler

        LEVEL: ERROR

        TIME: 21, May 2017 10:51

        MESSAGE: Duplicate records found in the user model
```
Comment (evansmurithi) [inline sms_log_handler/sms_handler.py]: Format the docstrings

**Review by evansmurithi — CHANGES_REQUESTED** [review]

File: sms_log_handler/providers/africastalking/gateway.py (hunk 3 above)
Comment (ikosenn) [reply sms_log_handler/providers/africastalking/gateway.py]: from where?

Comment (evansmurithi) [reply sms_log_handler/providers/africastalking/gateway.py]: https://github.com/AfricasTalkingLtd/africastalking-python

**Pull request merged**